### ISSUE TITLE
winsdk_version_conf with generator issue

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#15372]
Forgive me if I'm wrong: should this be `self._get_winsdk_version(system_version, generator_platform)` instead?
I have following conf in profile: 
```
tools.cmake.cmaketoolchain:generator=Visual Studio 17 2022
tools.cmake.cmaketoolchain:system_version=10.0.22000
```
and now I get `set(CMAKE_GENERATOR_PLATFORM "Visual Studio 17 2022,version=10.0.22000" CACHE STRING "" FORCE)` in `conan_toolchain.cmake`, which seems incorrect.

_Originally posted by @mocabe in https://github.com/conan-io/conan/pull/15272#discussion_r1438796086_

**Comment by memsharded** [comment]
This seems correct, I have managed to reproduce it, the problem was that this is only fired by the CMake new policy with latest CMake ``cmake_minimum_required(VERSION 3.27)``

**Comment by memsharded** [comment]
Thanks again for reporting @mocabe, the PR https://github.com/conan-io/conan/pull/15373 will fix it for next 2.0.17

**Comment by mocabe** [comment]
@memsharded 
Thank you for your quick response, I should have made a ticket first.
Hope 2024 will be a great year for the Conan project!

**Comment by AbrilRBS** [comment]
Closed by #15373, to be released in 2.0.17, thanks a lot for the report! :)
